crypto/cipher/cipher.c

```diff
@@ -171,7 +171,7 @@ int srtp_cipher_get_key_length(const srtp_cipher_t *c)
  */
 void srtp_cipher_rand_for_tests(void *dest, uint32_t len)
 {
-#if defined(HAVE_RAND_S)
+#if 0 // defined(HAVE_RAND_S) 
```

Comment:
Do we not need to check to see if this is available for some platforms?

Reply:
- The rand( ) function is part of the standard C library and should be available on all platforms, so the code in the #else part is always valid. However the rand( ) function may not give enough entropy and is otherwise also not very good because it returns limited values (as indicated by the remarks in the code).
- rand_s( ) is "better" than rand( ), it probably has better entropy (citation needed) and generates a wider variety of numbers. It also may return an error if there is something wrong. I'm not familiar with it (I use OpenSSL's randomizer for my applications) but I suspect that it returns an error if the random number generator isn't initialized or doesn't have enough entropy yet.
- The code in the #if section correctly checks for the error but incorrectly breaks off the process of filling the buffer with random values. It tries to return an error code but the function is declared void, so on Visual Studio that results in a compiler warning C4098 "void function returning a value".

So imagine the case that srtp_cipher_rand_for_tests( ) gets called and the code in the #if section is executed and srand_s returns an error. This results in the buffer not being filled or only being partially filled (the latter is unlikely but technically possible depending on srand_s's behavior). And functions that call srtp_cipher_rand_for_tests (and of course don't know that something went wrong) will get a buffer that may have "random" bytes but not as random as they wanted (basically the buffer will be unchanged from what it was before).

I concluded that:
- The behavior of the code in the #if block is wrong
- It's possible that the code in the #if block is not usually tested (is HAVE_RAND_S only defined for Windows builds?) or the warning about the void function returning a value is ignored by engineers
- The function is obviously used for testing only, and apparently the engineers who wrote the code and know more about intricacies of SRTP have decided that at least in SOME cases, testing with rand( ) would be sufficient
- Fixing the code "the right way" to eliminate the warning would require that I either (A) change the function's return type and change all the places where it's called to check the return value, or (B) change the code so that if rand_s fails, it would fall back to the rand( ) code. And if it would be sufficient to fall back to the rand( ) code, why not use that in the first place?
- This problem is ultimately unrelated to the problem that I'm trying to fix (compatibility with Visual Studio 2010) and I'm not familiar with the code so I didn't want the size of the change to go out of hand. 

The easiest way to work around the problem was to eliminate the #if section. I intended to explain this in my commit comment but I couldn't think of short way to explain my way of thinking.